src/modules/AutoinstFunctions.rb

```diff
@@ -121,6 +128,21 @@ def available_base_products
       end
     end
 
+    #
+    # Evaluate all available base products and returns a list of hashes which contains
+    # human readable strings only.
+    #
+    # @return [Hash] an array of product hashes
+    def available_base_products_hash
+      available_base_products.map do |product|
+        if Y2Packager::MediumType.offline? && !@force_libzypp
```

Comment:
I'd check the class of the objects, that's simpler and more reliable:

```suggestion
        if product.is_a?(Y2Packager::ProductLocation)
```